src/main/java/wingman/logic/flight/linklocation/FlightLocationLinkCommandFactory.java

```diff
@@ -0,0 +1,181 @@
+     * lazy and the flight manager lazy.
+     *
+     * @param locationManagerLazy the lazy instance of the location manager.
+     * @param flightManagerLazy   the lazy instance of the flight manager.
+     */
+    public FlightLocationLinkCommandFactory(
+            Lazy<ReadOnlyItemManager<Location>> locationManagerLazy,
+            Lazy<ReadOnlyItemManager<Flight>> flightManagerLazy,
+            FlightLocationLinkFunction<T> linkFunction,
+            String commandWord
+    ) {
+        super(flightManagerLazy, locationManagerLazy);
+        this.linkFunction = linkFunction;
+        this.commandWord = commandWord;
+    }
+
+    /**
+     * Returns a new link command factory.
+     *
+     * @return a new link command factory.
+     */
+    public static FlightLocationLinkCommandFactory<LinkFlightToLocationCommand> linkFactory() {
+        return new FlightLocationLinkCommandFactory<>(
+                LinkFlightToLocationCommand::new, LINK_COMMAND_WORD);
+    }
+
+    /**
+     * Returns a new unlink command factory.
+     *
+     * @return a new unlink command factory.
+     */
+    public static FlightLocationLinkCommandFactory<UnlinkFlightToLocationCommand> unlinkFactory() {
+        return new FlightLocationLinkCommandFactory<>(
+                UnlinkFlightToLocationCommand::new, UNLINK_COMMAND_WORD);
+    }
+
+    @Override
+    public String getCommandWord() {
+        return commandWord;
+    }
+
+    @Override
+    public Optional<Set<String>> getPrefixes() {
+        return Optional.of(Set.of(
+                FLIGHT_PREFIX,
+                LOCATION_DEPARTURE_PREFIX,
+                LOCATION_ARRIVAL_PREFIX
+        ));
+    }
+
+    @Override
+    public T createCommand(CommandParam param) throws ParseException,
+                                                      CommandException {
+        Optional<String> locationDepartureIdOptional =
+                param.getNamedValues(LOCATION_DEPARTURE_PREFIX);
+
+        Optional<String> locationArrivalIdOptional =
+                param.getNamedValues(LOCATION_ARRIVAL_PREFIX);
+
+        Flight flight = getSourceOrThrow(
+                FLIGHT_PREFIX,
+                param.getNamedValues(FLIGHT_PREFIX)
+        );
+
+        Map<FlightLocationType, Location> locationMap = new HashMap<>();
+
+        boolean hasLocation = addTarget(
+                LOCATION_DEPARTURE_PREFIX,
+                locationDepartureIdOptional,
+                FlightLocationType.LOCATION_DEPARTURE,
+                locationMap
+        ) | addTarget(
+                LOCATION_ARRIVAL_PREFIX,
+                locationArrivalIdOptional,
+                FlightLocationType.LOCATION_ARRIVAL,
+                locationMap
+        );
+
+        if (!hasLocation) {
+            throw ParseException.formatted(
+                    NO_LOCATION_MESSAGE,
+                    commandWord,
+                    getCommandFormatHint()
+            );
+        }
```

Comment:
Seems duplicated across different `LinkCommandFactory`s, but we can just leave them, given there are some class-specific differences here, e.g., `LOCATION_ARRIVAL_PREFIX` is unique to the Flight-Location link.